### ISSUE TITLE
KeyError: '0'

### ISSUE THREAD

**Issue by mlindauer** [automl/CAVE#114]
After fixing the path issue, I have now a new issue:

```
$ python ../../../../../scripts/cave.py --folders smac3-output_2018-01-11_15\:43\:56_918369/* --validation epm
INFO:cave.cavefacade:Saving results to CAVE_output
INFO:cave.SMACrun.smac3-output_2018-01-11_15:43:56_918369/run_3:('smac3-output_2018-01-11_15:43:56_918369', 'run_3')
INFO:cave.cavefacade:Completing data using epm.
INFO:smac.utils.validate.Validator:Collected 8 runs from 2 configurations on 11 instances with 1 repetitions. Reusing 14 runs from given runhistory.
Traceback (most recent call last):
  File "../../../../../scripts/cave.py", line 17, in <module>
    cave.main_cli()
  File "/home/lindauer/git/SpySMAC3/cave/cave_cli.py", line 101, in main_cli
    fanova_pairwise=args_.fanova_pairwise)
  File "/home/lindauer/git/SpySMAC3/cave/cavefacade.py", line 144, in __init__
    self.complete_data(method=missing_data_method)
  File "/home/lindauer/git/SpySMAC3/cave/cavefacade.py", line 184, in complete_data
    runhistory=self.original_rh)
  File "/home/lindauer/anaconda3/lib/python3.6/site-packages/smac/utils/validate.py", line 319, in validate_epm
    self.scen.feature_dict[run.inst]])
KeyError: '0'
```

Data is attached
[smac3.tar.gz](https://github.com/automl/CAVE/files/1646489/smac3.tar.gz)

**Comment by AndreBiedenkapp** [comment]
This seems to be an error from SMAC3.
The issue seems to be with the validator here:
https://github.com/automl/SMAC3/blob/2ec1fa9dd5403b189a3a79c20c4cc2774b2d8fd6/smac/utils/validate.py#L317-L319
The validator checks if the scenario object has a feature_dictionary. In your data you have training and test instances, but no features. So there is a feature dictionary that basically only consists of keys, but no values.

I'm not sure if this is should be fixed in SMACs scenario or in the validator but I don't think this is an issue with CAVE.

**Comment by shukon** [comment]
[Submitted issue](https://github.com/automl/SMAC3/issues/377). Will fix now.

**Comment by shukon** [comment]
Fixed in development-branch of SMAC, resolved in [#378](https://github.com/automl/SMAC3/pull/378)
